app/src/main/java/com/ncnf/database/DatabaseService.java

```diff
@@ -1,125 +1,245 @@
 package com.ncnf.database;
 
-        import android.os.Build;
-        import android.util.Log;
-
-        import androidx.annotation.RequiresApi;
-
-        import com.google.android.gms.tasks.SuccessContinuation;
-        import com.google.android.gms.tasks.Task;
-        import com.google.firebase.auth.AuthResult;
-        import com.google.firebase.firestore.CollectionReference;
-        import com.google.firebase.firestore.DocumentSnapshot;
-        import com.google.firebase.firestore.FieldPath;
-        import com.google.firebase.firestore.FieldValue;
-        import com.google.firebase.firestore.FirebaseFirestore;
-        import com.google.firebase.firestore.Query;
-        import com.ncnf.authentication.AuthenticationResponse;
-
-        import java.util.ArrayList;
-        import java.util.List;
-        import java.util.Map;
-        import java.util.concurrent.CompletableFuture;
-        import java.util.concurrent.FutureTask;
-
-        import javax.inject.Inject;
-
-        import static com.ncnf.Utils.*;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FieldPath;
+import com.google.firebase.firestore.FieldValue;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.ncnf.database.builder.DatabaseObjectBuilder;
+import com.ncnf.database.builder.EventBuilder;
+import com.ncnf.database.builder.UserBuilder;
+import com.ncnf.event.Event;
+import com.ncnf.event.PrivateEvent;
+import com.ncnf.event.PublicEvent;
+import com.ncnf.user.User;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
 
 public class DatabaseService implements DatabaseServiceInterface {
 
     private final FirebaseFirestore db;
+    private final Map<Class, DatabaseObjectBuilder> registry = new HashMap<>();
 
     @Inject
     public DatabaseService(){
         this.db = FirebaseFirestore.getInstance();
+        initMap();
```

Comment:
Minor detail, but you could call this function something like initRegistry()

Reply:
Done !